block-report.js

```diff
@@ -0,0 +1,151 @@
+
+const PROJECT = 'bacom-blog';
+const SITE = 'https://main--business-website--adobe.hlx.page';
+const INDEX = '/blog/query-index.json?limit=3000';
+const USE_CACHE = true;
+
+async function fetchIndex(url) {
+    console.log('Fetching entries and saving locally');
+    const index = await fetch(url);
+    const indexData = await index.json();
+
+    return indexData.data;
+}
+
+async function loadIndex(project, url, cached = true) {
+    const filePath = `./${project}/entries.json`;
+
+    async function fetchAndWrite() {
+        const entries = await fetchIndex(url);
+        const paths = entries.map((entry) => entry.path);
+        await writeFile(filePath, JSON.stringify(paths, null, 2));
+        return paths;
+    }
+
+    if (!cached) {
+        return fetchAndWrite();
+    }
+
+    try {
+        await access(filePath);
+        const entriesJson = await readFile(filePath, 'utf8');
+        return JSON.parse(entriesJson);
+    } catch (err) {
+        return fetchAndWrite();
+    }
+}
+
+async function fetchMarkdown(url) {
+    console.log('Fetching markdown and saving locally');
+    const response = await fetch(url);
+    return await response.text();
+}
+
+async function loadMarkdown(url, path, cached = true) {
+    async function fetchAndWrite() {
+        const markdown = await fetchMarkdown(`${url}.md`);
+        await writeFile(`./${path}.md`, markdown);
+        return markdown;
+    }
+
+    if (!cached) {
+        return fetchAndWrite();
+    }
+
+    try {
+        await access(`./${path}.md`);
+        return await readFile(`./${path}.md`, 'utf8');
+    } catch (err) {
+        return fetchAndWrite();
+    }
+}
+
+async function runReport(project, site, indexUrl, cached = true) {
+    const blocks = {};
+    const variants = {};
+    const totals = { blocks: {}, variants: {} };
+    const entries = await loadIndex(project, indexUrl, cached);
+
+    for (const entry of entries) {
+        blocks[entry] = {};
+        const folder = `${project}${entry.split('/').slice(0, -1).join('/')}`;
+        await mkdir(`./${folder}`, { recursive: true });
+
+        const markdown = await loadMarkdown(`${site}${entry}`, `${project}${entry}`, cached);
+        const mdast = await getMdast(markdown);
+        const tableMap = getTableMap(mdast);
+
+        tableMap.forEach((table) => {
+            const { blockName, options } = table;
+            if (options) {
+                const variant = blockName + ' (' + options.join(', ') + ')';
+                variants[entry] = variants[entry] || {};
+                variants[entry][variant] = (variants[entry][variant] || 0) + 1;
+                totals.variants[variant] = (totals.variants[variant] || 0) + 1;
+            }
+            blocks[entry][blockName] = (blocks[entry][blockName] || 0) + 1;
+            totals.blocks[blockName] = (totals.blocks[blockName] || 0) + 1;
+        });
+
+
+        console.log(entry, blocks[entry]);
```

Comment:
```suggestion
        console.log(entry, blocks[entry]);
        if (i < entries.length - 1) {
            await new Promise((resolve) => setTimeout(resolve, 500));
        }        
```

Reply:
Sounds good, however I'll add the delay in `fetchMarkdown` so to no slow down cached content.